.maintain/chaostest/src/config/index.js

```diff
@@ -0,0 +1,67 @@
+const fs = require('fs')
+const path = require('path')
+const configPath = path.join(__dirname, './config.json')
+
+class Config {
+    constructor() {
+        this.load()
+    }
+    
+    async load() {
+        fs.readFile(configPath, (err, data) => {
+            if (err) {
+                if (err.code === 'ENOENT') {
+                    this.reset()
+                } else {
+                    throw err
+                }
+            }  else {
+                try {
+                    Object.assign(this, JSON.parse(data));
+                } catch (error) {
+                    console.log(`config file is corrupted, resetting...`)
+                    this.reset()
+                }
+            };
+        });
+    };
+
+    getConfig () {
+        return this
+    }
+
+    async update() {
+        let data = JSON.stringify(this.getConfig());
+        fs.writeFile(configPath, data, (err) => {
+            if (err) throw err;
+            console.log('Configuration updated');
+        });
+    }
+
+    async setNameSpace(namespace) {
+        this.namespace = namespace
+        this.update()
+    }
+
+    async addNode(node) {
+        if (!this.nodes || Array.isArray(this.nodes)) {
+            this.nodes = []
+        }
+        if (node.nodeType === 'bootnode') {
+            this.bootnode = node
+        }
+        this.nodes.push(node)
+        this.update()
+    }
+
+    async reset() {
+        let data = JSON.stringify({});
+        fs.writeFile(configPath, data, (err) => {
+            if (err) throw err;
+            this.load()
+        });
+    }
+}
+
+const config = new Config()
```

Comment:
This is a global variable, right? I would suggest not to use any global variables as they make code difficult to reason about, test and debug.

Reply:
> This is a global variable, right? I would suggest not to use any global variables as they make code difficult to reason about, test and debug.

I haven't thought of a good way to work around this.

Reply:
Why don't you pass it as an argument to the functions that need it?

Reply:
I am feeling right to pass it to any functions as an argument. Maybe i will make ```Deployment``` a class and make it a private property. Allow me to proper fix that in next feature update.

Reply:
I am not in favor of starting off this project with technical dept on the very first commit. Do you need help on how one would achieve this without global variables?

Reply:
Bumping this conversation.